test/data/spreadsheets-health.test.ts

```diff
@@ -23,22 +28,55 @@ test(
         });
 
         const output = await spreadsheetToJson(state, rows, false, null);
-        if (output.invalidCollectedIncentives.length > 0) {
-          console.error(
-            `${state}` +
-              util.inspect(
-                output.invalidCollectedIncentives.map(invalid => [
-                  invalid.id,
-                  invalid.errors,
-                ]),
-                { depth: null },
-              ),
-          );
-        }
-        tap.equal(
-          output.invalidCollectedIncentives.length,
-          0,
-          `${state}'s spreadsheet does not match the collected data schema`,
+        const invalidCollectedPath = file.filepath.replace(
+          '.json',
+          '_invalid_collected.json',
+        );
+        const invalidStatePath = file.filepath.replace(
+          '.json',
+          '_invalid_state.json',
+        );
+        const previousCollectedExceptions: string = fs.existsSync(
+          invalidCollectedPath,
+        )
+          ? JSON.parse(fs.readFileSync(invalidCollectedPath, 'utf-8'))
+          : [];
+        const previousStateExceptions: string = fs.existsSync(invalidStatePath)
+          ? JSON.parse(fs.readFileSync(invalidStatePath, 'utf-8'))
+          : [];
+
+        // We expect equality of any invalid records, since humans shouldn't be
+        // editing these anyway.
```

Comment:
Isn't it possible that new, invalid records have been added to the spreadsheet? I agree that humans shouldn't be editing the invalid records.

Reply:
If so, then those records should be added to the golden checked-in file.

The comment is supposed to contrast from the approach further below where we only test equality of some fields, but let me know if it's misleading.